### ISSUE TITLE
ExecStart=/usr/bin/dockerd -H fd:// (code=exited, status=1/FAILURE) __ Docker Installation Failed  

### ISSUE THREAD

**Issue by alimp5** [docker/docker.github.io#7170]
### Problem description
Greetings!
I got the last released of Docker for Debian[Kali Linux] from below link:
https://download.docker.com/linux/debian/dists/jessie/pool/stable/amd64/

Debian link: https://download.docker.com/linux/debian/dists/jessie/pool/stable/amd64/docker-ce_18.06.0~ce~3-0~debian_amd64.deb)

```
root@kalia:~# uname -a
Linux kalia.ir 4.17.0-kali1-amd64 #1 SMP Debian 4.17.8-1kali1 (2018-07-24) x86_64 GNU/Linux


root@kalia:~# dpkg -i docker-ce_18.06.0~ce~3-0~debian_amd64.deb 
Selecting previously unselected package docker-ce.
(Reading database ... 643662 files and directories currently installed.)
Preparing to unpack docker-ce_18.06.0~ce~3-0~debian_amd64.deb ...
Unpacking docker-ce (18.06.0~ce~3-0~debian) ...
Setting up docker-ce (18.06.0~ce~3-0~debian) ...
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
invoke-rc.d: initscript docker, action "start" failed.
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: exit-code) since Sat 2018-08-04 15:15:55 +0430; 6ms ago
     Docs: https://docs.docker.com
  Process: 13254 ExecStart=/usr/bin/dockerd -H fd:// (code=exited, status=1/FAILURE)
 Main PID: 13254 (code=exited, status=1/FAILURE)
dpkg: error processing package docker-ce (--install):
 installed docker-ce package post-installation script subprocess returned error exit status 1
Processing triggers for systemd (239-7) ...
Processing triggers for man-db (2.8.3-2) ...

Errors were encountered while processing:
 docker-ce
```
How to fix the problem !?

**Comment by alimp5** [comment]
Even i removed and purged the downloaded + installed (failed) Docker and tried to install docker via built-in Debian repositories. but again failed.

output:
```
root@kalia:~# apt-get install docker-ce 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  asciidoctor augeas-lenses bsdtar caja-common certbot db-util dissy gcc-7-base:i386
  geoip-database-extra ..................<skipped>.................

Use 'apt autoremove' to remove them.
The following NEW packages will be installed:
  docker-ce
0 upgraded, 1 newly installed, 0 to remove and 99 not upgraded.
Need to get 39.8 MB of archives.
After this operation, 201 MB of additional disk space will be used.
Get:1 https://download.docker.com/linux/debian jessie/stable amd64 docker-ce amd64 18.06.0~ce~3-0~debian [39.8 MB]
Fetched 39.8 MB in 17s (2,343 kB/s)                                                                            
Selecting previously unselected package docker-ce.
(Reading database ... 643658 files and directories currently installed.)
Preparing to unpack .../docker-ce_18.06.0~ce~3-0~debian_amd64.deb ...
Unpacking docker-ce (18.06.0~ce~3-0~debian) ...
Setting up docker-ce (18.06.0~ce~3-0~debian) ...
update-rc.d: We have no instructions for the docker init script.
update-rc.d: It looks like a non-network service, we enable it.
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
invoke-rc.d: initscript docker, action "start" failed.
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: activating (auto-restart) (Result: exit-code) since Sat 2018-08-04 15:31:03 +0430; 6ms ago
     Docs: https://docs.docker.com
  Process: 15457 ExecStart=/usr/bin/dockerd -H fd:// (code=exited, status=1/FAILURE)
 Main PID: 15457 (code=exited, status=1/FAILURE)
    Tasks: 40 (limit: 4915)
   Memory: 3.2M
   CGroup: /system.slice/docker.service

Aug 04 15:31:03 kalia.ir systemd[1]: Failed to start Docker Application Container Engine.
dpkg: error processing package docker-ce (--configure):
 installed docker-ce package post-installation script subprocess returned error exit status 1
Processing triggers for systemd (239-7) ...
Processing triggers for man-db (2.8.3-2) ...
Errors were encountered while processing:
 docker-ce
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@kalia:~# systemctl status docker.service
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Sat 2018-08-04 15:31:05 +0430; 22s ago
     Docs: https://docs.docker.com
  Process: 15983 ExecStart=/usr/bin/dockerd -H fd:// (code=exited, status=1/FAILURE)
 Main PID: 15983 (code=exited, status=1/FAILURE)

Aug 04 15:31:04 kalia.ir systemd[1]: docker.service: Failed with result 'exit-code'.
Aug 04 15:31:04 kalia.ir systemd[1]: Failed to start Docker Application Container Engine.
Aug 04 15:31:05 kalia.ir systemd[1]: docker.service: Service RestartSec=100ms expired, scheduling restart.
Aug 04 15:31:05 kalia.ir systemd[1]: docker.service: Scheduled restart job, restart counter is at 4.
Aug 04 15:31:05 kalia.ir systemd[1]: Stopped Docker Application Container Engine.
Aug 04 15:31:05 kalia.ir systemd[1]: docker.service: Start request repeated too quickly.
Aug 04 15:31:05 kalia.ir systemd[1]: docker.service: Failed with result 'exit-code'.
Aug 04 15:31:05 kalia.ir systemd[1]: Failed to start Docker Application Container Engine.
root@kalia:~# 
```

**Comment by thaJeztah** [comment]
make sure to check the full logs for the docker service (using `journalctl`); you'll likely find the actual error message in there

**Comment by L-Hudson** [comment]
Closing this ticket due to its age, and the impending refactor. If you think this is in error, feel free to reopen. Thanks!

**Comment by karlwilbur** [comment]
I know this isn't a `docker.github.io` issue, but I just wanted to throw this out there for anyone else that might come across this thread looking for a solution. 

I had the same problem. I had Docker running fine on my other machines (laptops and desktops). I was, at the time, running Ubuntu 16.04 on all of my machines.

I could not get the `dockerd` service to start successfully on my shiny new 16.04 laptop. I would get the same error as the OP here.
```
ExecStart=/usr/bin/dockerd -H fd:// (code=exited, status=1/FAILURE)
```

I spent hours trying different things with no luck. It turned out it was because my VPN (ExpressVPN) was running. I disconnected the VPN and everything was fine. 

I found that it was a networking issue because I ran:
```bash
sudo dockerd -D
```

Hope this helps.
